homeassistant/components/webdav/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The WebDAV integration."""
+
+from __future__ import annotations
+
+import logging
+
+from aiowebdav2.client import Client
+from aiowebdav2.exceptions import UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+
+from .const import CONF_BACKUP_PATH, DATA_BACKUP_AGENT_LISTENERS
+from .helpers import async_create_client, async_ensure_path_exists
+
+type WebDavConfigEntry = ConfigEntry[Client]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: WebDavConfigEntry) -> bool:
+    """Set up WebDAV from a config entry."""
+    client = async_create_client(
+        hass=hass,
+        url=entry.data[CONF_URL],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        verify_ssl=entry.data.get(CONF_VERIFY_SSL, True),
+    )
+
+    try:
+        result = await client.check()
+    except UnauthorizedError as err:
+        raise ConfigEntryError("Invalid username or password") from err
+
+    # Check if we can connect to the WebDAV server
+    # and access the root directory
+    if not result:
+        raise ConfigEntryNotReady("Failed to connect to WebDAV server")
+
+    # Ensure the backup directory exists
+    if not await async_ensure_path_exists(
+        client, entry.data.get(CONF_BACKUP_PATH, "/")
+    ):
+        raise ConfigEntryNotReady("Failed to create backup directory")
```

Comment:
Is it expected that this will be successful later without any user interaction? If not we should raise `ConfigEntryError` so the integration doesn't try to reload all the time, but shows a proper error message in the frontend

Reply:
after thinking about it some more, `ConfigEntryNotReady` might be ok, but make it [translatable](https://developers.home-assistant.io/docs/internationalization/core/#exceptions) - the better approach would be implementing an repair issue, which explains the user what happened and how to solve the issue

Reply:
> the better approach would be implementing an repair issue, which explains the user what happened and how to solve the issue

Agreed, but I thinks this is nothing for a initial implementation :)